_posts/2017-07-29-ruby-oop.md

```diff
@@ -0,0 +1,383 @@
+---
+layout: post
+author: z64
+title: Beginning OOP with Ruby
+date: 2017-07-29
+tags: ruby oop tutorial
+---
+
+This post attempts to guide a beginner through object-oriented progamming in Ruby, showing them the thought process behind both designing and interacting with a library, and problem solving in general with Ruby.
+
+To accomplish this, we will be solving a simple geometry problem:
+
+- Describe a collection of objects that can calculate the perimeter of a polygon with an arbitrary number of sides.
+
+## Breaking it down
+
+The first thing we want to do to tackle this problem is to break apart the problem into a series of smaller, bite sized questions. "Good Ruby code" will read a lot like English - I think you'll find the syntax we end up writing will almost look like English answers to these questions. You probably already know the answers to these (which is why I picked this as an example), but read them and pay attention to the language that I use.
+
+**1) What is perimeter?**
+
+> Perimeter is the total distance around an object, in our case a 2D polygon with any number of sides.
+
+**2) What is a Polygon?**
+
+> A **polygon** is an object **made up of lines**.
+
+**3) Okay, so what is a line?**
+
+> A line is the shortest distance between **two points** in space.
+Since we're dealing with finite lines, they have a *start* and an *end*.
+
+**4) Okay, so what is a point?**
+
+> A point is a location in space. For our purposes, this location is on a 2D plane, bound by **coordinates**.
+
+In the next section, we'll convert these concepts into Ruby code. Please re-read this section once we're done and see how what we said here is directly represented in the answer we come up with.
```

Comment:
Remove the please to make it more direct, and possibly add a part at the bottom that says to go back up and read this part. Maybe with a link too. Or move this second part to the end of the post.